🤖 Update labels

ErikSchierboom opened the pull request:
This PR adds:

- A `.github/labels.yml` file containing this repository's labels
- An `.appends/.github/labels.yml` file, which contains all the repository-specific labels currently used in this repo.
- A `.github/workflows/sync-labels.yml` workflow to automatically synchronize this repository's labels (unless alreadt present)

With this setup, the labels that can be used in this repo are all defined in a single file: the `.github/labels.yml` file. The `sync-labels.yml` workflow automatically runs whenever this file changes, and will update the repository's label to match the labels defined in the `.github/labels.yml` file. Note that is will remove any labels _not_ in the `.github/labels.yml` file, so be careful with removing labels from the `.github/labels.yml` file.

The `.github/labels.yml` file is auto-generated by concatenating these two files:

- The Exercism-wide labels defined in https://github.com/exercism/org-wide-files/blob/main/global-files/.github/labels.yml
- The repo-specific labels defined in `.appends/.github/labels.yml`

Whenever one of these two files change, a pull request is automatically submitted to update the `.github/labels.yml`  file. Merging that pull request will then trigger the sync-labels workflow, and the labels will be updated.

With this setup, we are able to guarantee that each repository can use both the Exercism-wide labels and any track-specific labels.

## Tracking

https://github.com/exercism/v3-launch/issues/41

SaschaMann commented on .appends/.github/labels.yml:
where does this come from?

SaschaMann reviewed (commented):
rest looks good

ErikSchierboom committed:
Define repo-specific labels.

This commit adds a `.appends/.github/labels.yml` file, which contains the repo-specific labels. This file will automatically be combined with the Exercism-wide labels defined in https://github.com/exercism/org-wide-files/blob/main/global-files/.github/labels.yml to form the `.github/labels.yml` file.

ErikSchierboom committed:
Add a GitHub Actions workflow to automatically sync the repository labels.

This commit adds a `.github/workflow/sync-labels.yml` file, which defines a workflow that syncs this repository's labels with the contents of the `.github/labels.yml` file. The labels are synced automatically whenever the `.github/labels.yml` file changes.

ErikSchierboom committed:
Define the labels used in this repo.

This commit adds a `.github/labels.yml` file, which contains the full list of labels that this repo can use. This file is a combination of the `.appends/.github/labels.yml` file and the Exercism-wide labels defined in https://github.com/exercism/org-wide-files/blob/main/global-files/.github/labels.yml.
